client/fetch_state.go

```diff
@@ -48,7 +48,7 @@ type fetchState struct {
 	sync.Mutex
 	refCounter
 
-	op                   *fetchTaggedOp
+	op                   fetchTaggedOp
```

Comment:
Just out curiousity, what was the impetus to make this an interface (does it need mocking?) - seems `fetchState` knowing the concrete type should be fine no?

Reply:
Ah I see the mocking below.  If we have to mock and there's significant benefits, then we gotta do what we gotta do.

Reply:
cheers, managed to use the concrete type for `fetchTaggedOp` too.